From 6116b541a3d90a4fd00760dc0273f9c3cb5d324a Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Fri, 22 Jul 2016 22:58:54 +0000
Subject: [PATCH 1/5] Documentation: dt: rng: add bindings doc for Keystone SA
 HWRNG driver

The Keystone SA module has a hardware random generator module.
This commit adds binding doc for the KS2 SA HWRNG driver.

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 .../devicetree/bindings/rng/ks-sa-rng.txt          | 33 ++++++++++++++++++++++
 1 file changed, 33 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/rng/ks-sa-rng.txt

diff --git a/Documentation/devicetree/bindings/rng/ks-sa-rng.txt b/Documentation/devicetree/bindings/rng/ks-sa-rng.txt
new file mode 100644
index 0000000..af228f5
--- /dev/null
+++ b/Documentation/devicetree/bindings/rng/ks-sa-rng.txt
@@ -0,0 +1,33 @@
+KEYSTONE SoC HWRNG Module
+
+On Keystone SoCs HWRNG module is a submodule of the Security Accelerator.
+
+- compatible: should be "ti,keystone-rng"
+- sa-cfg: phandle to syscon node of the SA configuration registers.
+	  This registers are shared between hwrng and crypto drivers.
+- clocks: phandle to the reference clocks for the subsystem
+- clock-names: functional clock name. Should be set to "fck"
+- reg:    HWRNG module register space
+
+Example:
+/* K2HK */
+
+sa_subsys: subsys@20c0000 {
+	#address-cells = <1>;
+	#size-cells = <1>;
+	ranges = <0 0x20c0000 0x40000>;
+	compatible = "simple-bus";
+
+	sa_config: subsys {
+		compatible      = "syscon";
+		reg             = <0x0 0x100>;
+	};
+
+	hwrng {
+		compatible = "ti,keystone-rng";
+		sa-cfg = <&sa_config>;
+		clocks = <&clksa>;
+		clock-names = "fck";
+		reg = <0x24000 0x1000>;
+	};
+};
-- 
1.9.1

